packages/dcos-net/extra/dcos-net-setup.py

```diff
@@ -30,38 +31,53 @@ def main():
         sys.stderr.buffer.write(result.stderr)
         if result.stderr.strip().endswith(b'File exists'):
             result.returncode = 0
-    elif sys.argv[1] == 'iptables':
+        sys.exit(result.returncode)
+
+    if sys.argv[1] == 'iptables':
         # check whether a rule matching the specification does exist
         argv = ['-C' if arg in ['-A', '-I'] else arg for arg in sys.argv[1:]]
         result = subprocess.run(argv)
         if result.returncode != 0:
             # if it doesn't exist append or insert that rules
             result = subprocess.run(sys.argv[1:])
-    elif sys.argv[1] == '--ipv6':
+        sys.exit(result.returncode)
+
+    if sys.argv[1] == '--ipv6':
         if os.getenv('DCOS_NET_IPV6', 'true') == 'false':
             sys.exit(0)
         else:
             del sys.argv[1]
             result = subprocess.run(sys.argv)
-    elif sys.argv[1:3] == ['networkd', 'add'] and len(sys.argv) == 4:
-        result = add_networkd_config(sys.argv[3])
-    else:
-        result = subprocess.run(sys.argv[1:])
-    sys.exit(result.returncode)
+        sys.exit(result.returncode)
+
+    if sys.argv[1:3] == ['networkd', 'add'] and len(sys.argv) == 4:
+        result_code = add_networkd_config_for_coreos(sys.argv[3])
+        sys.exit(result_code)
+
+    result = subprocess.run(sys.argv[1:])
 
 
-def add_networkd_config(src):
+def add_networkd_config_for_coreos(src: str) -> int:
+    # systemd-netword, when enabled, will wipe the configurations like IP
+    # address of network interfaces and this behavior happens only on coreos
+    # This problem is tracked by:
+    # https://jira.mesosphere.com/browse/DCOS_OSS-1790
+    # We need to mark interfaces managed by DC/OS as unmanaged when networkd is
+    # enabled on coreos
+    if platform.system() != "Linux" or "coreos" not in platform.release():
+        return 0
+
     networkd = b'systemd-networkd.service'
     networkd_path = '/etc/systemd/network'
 
     # Check if there is networkd
-    result = subprocess.run(['systemctl', 'list-unit-files', networkd],
-                            stdout=subprocess.PIPE)
-    if result.returncode != 0:
-        return result
-    if networkd not in result.stdout:
-        return result
+    list_unit_ret = subprocess.run(['systemctl', 'list-unit-files', networkd],
```

Comment:
`list_unit_files_ret`?

Reply:
Or simply `result`?

Reply:
We have several different commands, and it's a better practice to give the result different names in case of misuse, what do you think?

Reply:
AFAIK, there is a best practice for every single way of doing things. To me, it is a matter of writing a blog post and calling it a best practice :)

Personally, I don't think it improves readability and/or makes the code safer. Just my $.02, but it is your call.

Reply:
I am on the fence when I see changes like these. Personally, I like to see minimal code changes as long as there isn't a pressing need to make other changes. Remember any code that you touch has the potential of introducing a regression. Having said that I would be inclined to use the previous code

Reply:
I did not say it's the best practice, see the code below
```
ret = a + b
if ret > 10:
    return ret

retu = c + d
if ret > 10:
    return ret
```

Reply:
I reverted the code.